lib/parse.js

```diff
@@ -84,6 +135,35 @@ var INLINE_CODE = 'inlineCode';
 var BREAK = 'break';
 var ROOT = 'root';
 
+/** TODO */
+function isWhiteSpace(character) {
```

Comment:
This is equivalent to `/\s/.test`, right? I measured it on Node (got nearly the same results, regexp version is slightly faster), on Firefox (regexp version is 2x faster) and on (older) Chromium (`isWhiteSpace` is faster by ~25%). I also created a [jsperf][2] for this.

It is not clear to me why `isWhiteSpace` version should be faster. JavaScript engines adopted JIT compilation for regular expressions [long time ago][1]. JITed code will be basically equivalent to this if-else chain, it can only be better and shared with other parts of the program. Expanding RegExp character classes manually to if-then chains like this is hardly an algorithmic improvement.

I just hope the main reason behind this version of `isWhiteSpace` is that it is more readable and maintainable this way. It could be.

[1]: http://blog.chromium.org/2009/02/irregexp-google-chromes-new-regexp.html
[2]: http://jsperf.com/iswhitespace-vs-regexp/4